### DISCUSSION TITLE
Low Ingestion Performance in Apache Doris Compared to Elasticsearch

### DISCUSSION THREAD

**Discussion by Rajeshpahi** [apache/doris#36137] [A - General / Q&A]
Hello Apache Doris Community,

I am experiencing significantly lower performance with Apache Doris compared to Elasticsearch during data ingestion and am seeking advice on how to address this issue.

System Configuration:

Hardware:	Ram= 24GB
CPU = 4 Core
HDD = 1 TB

OS Details:	Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal

Versions:

Apache Doris: 2.1.2
Elasticsearch: 7.4.2

Data Ingestion Details:

Data Volume: 0.87 GB
Data Format: .log
Ingestion Rate: Batch-size is set to 40,000 logs per second
Sample Log: “10.10.30.77 - - [2024-05-20T06:21:25] "GET /DashboardServer/v2/web/common/generateNewToken?validityPeriod=2160%3A00%3A00 HTTP/1.1" 200 30482 328 https-jsse-nio-0.0.0.0-7901-exec-459”

Performance Details:

-Apache-Doris is taking 11 minutes to ingest 4,800,000 document from logstash with simulation rate of 40,000 log per second, while elasticsearch is taking 07 minutes to ingest same document from logstash with same configuration.

Best regards,
Rajesh Pahi

**Comment by xiaokang** [comment]
@Rajeshpahi Can you provide more information to reproduce the comparison? 

1. Doris table schema and ES index mapping
2. logstash configuration
3. log simulation code

**Reply by Rajeshpahi** [reply]
1. **Doris table schema**
CREATE TABLE aa (
id BIGINT NOT NULL AUTO_INCREMENT,
timestamp DATETIME,
message TEXT,
INDEX idx_comment (message) USING INVERTED PROPERTIES("parser" = "english") COMMENT 'inverted index for comment'

)
DUPLICATE KEY(id)

DISTRIBUTED BY HASH(id)
BUCKETS 10
PROPERTIES ("replication_num" = "1");

2. **Logstash-filter Configuration**:-
input {
file {
path => "/home/logs/sim/real_log/db1.log"
start_position => "beginning"
sincedb_path => "/dev/null"
}
}

filter {
grok {
match => { "message" => "(.)-\s[(?[^\]]+)]\s+(?.)" }
}

date {
match => [ "timestamp", "yyyy-MM-dd'T'HH:mm:ss" ]
}

mutate {
remove_field => [ "host", "path", "@Version", "@timestamp" ]
}
}

output {
http {
follow_redirects => true
url => "http://doris-BE host:doris-BE port/api/database_name/table_name/_stream_load"
http_method => "put"
content_type => "application/json"
ignorable_codes => [307]
headers => {
"strip_outer_array" => "true"
"format" => "json"
"Expect" => "100-continue"
"Authorization" => "Basic YWRtaW46YWRtaW4="
}
format => "json_batch"
#batch_size => 150
}
}

3. **log simulation code**:-
We have used Java simulator which generates continuous logs per second with a certain ingestion rate and appended the data into .log file which is used for ingestion in Apache-Doris.

[Logstash_yml.txt](https://github.com/user-attachments/files/15815374/Logstash_yml.txt)
[ES_mapping.txt](https://github.com/user-attachments/files/15815375/ES_mapping.txt)

**Reply by Rajeshpahi** [reply]
Hi Team,
Any update on above comment?
